src/credential/issuance/03-start-credential-issuance.ts

```diff
@@ -0,0 +1,336 @@
+    idphint: string;
+    appFetch?: GlobalFetch["fetch"];
+  }
+) => Promise<CredentialResponse>;
+
+/**
+ * Starts the credential issuance flow to obtain a credential from the issuer.
+ * @param issuerConf The Issuer configuration
+ * @param credentialType The type of the credential to be requested
+ * @param context.wiaCryptoContext The context to access the key associated with the Wallet Instance Attestation
+ * @param context.credentialCryptoContext The context to access the key to associat with credential
+ * @param context.identificationContext The context to identify the user which will be used to start the authorization
+ * @param context.walletInstanceAttestation The Wallet Instance Attestation token
+ * @param context.redirectUri The internal URL to which to redirect has passed the in-app browser login phase
+ * @param context.idphint Unique identifier of the SPID IDP
+ * @param context.appFetch (optional) fetch api implementation. Default: built-in fetch
+ * @throws {IdentificationError} When the response from the identification response is not parsable
+ * @returns The credential obtained
+ */
+
+export const startCredentialIssuance: StartCredentialIssuance = async (
+  issuerConf,
+  credentialType,
+  ctx
+) => {
+  const {
+    wiaCryptoContext,
+    credentialCryptoContext,
+    identificationContext,
+    walletInstanceAttestation,
+    redirectUri,
+    idphint,
+    appFetch = fetch,
+  } = ctx;
+
+  /**
+   * Creates and sends a PAR request to the /as/par endpoint of the authroization server.
+   * This starts the authentication flow to obtain an access token.
+   * This token enables the Wallet Instance to request a digital credential from the Credential Endpoint of the Credential Issuer.
+   * This is an HTTP POST request containing the Wallet Instance identifier (client id), the code challenge and challenge method as specified by PKCE according to RFC 9126
+   * along with the WTE and its proof of possession (WTE-PoP).
+   * Additionally, it includes a request object, which is a signed JWT encapsulating the type of digital credential requested (authorization_details),
+   * the application session identifier on the Wallet Instance side (state),
+   * the method (query or form_post.jwt) by which the Authorization Server
+   * should transmit the Authorization Response containing the authorization code issued upon the end user's authentication (response_mode)
+   * to the Wallet Instance's Token Endpoint to obtain the Access Token, and the redirect_uri of the Wallet Instance where the Authorization Response
+   * should be delivered. The redirect is achived by using a custom URL scheme that the Wallet Instance is registered to handle.
+   */
+  const clientId = await wiaCryptoContext.getPublicKey().then((_) => _.kid);
+
+  // WARNING: This is not a secure way to generate a code verifier CHANGE ME
+  const codeVerifier = generateRandomAlphaNumericString(64);
```

Comment:
Do we want to use a TODO instead of the Warning?

Reply:
Furthermore, I would add the todo in the `generateRandomAlphaNumericString` function and not here

Reply:
Done in 4f72ca9c4f3ae3a04ceb582794f909e3d907fd87. Apparently even `react-native-uuid` is based on `Math.random` so that has to be replace as well: